assets/aml-benchmark/components/dataset-preprocessor/spec.yaml

```diff
@@ -0,0 +1,56 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+type: command
+
+name:  dataset_preprocessor 
+display_name: Dataset Preprocessor
+description: Dataset Preprocessor
+version: 0.0.1
+is_deterministic: true
+
+inputs:
+  dataset: 
+    type: uri_file
+    description: >-
+      Path to load the dataset.
+    optional: False
+  template_input:
+    type: string
+    description: >-
+      A json dictionary where key is the name of the column enclosed in " " and associated dict value is presented using jinja template logic 
+      which will be used to extract respective value from the dataset. 
+      Example format:
+      {"<user_column_name>": {{key in the json file for this column}}, ....}. The processed output will be dumped to a 
+      jsonl file in this format: {"<user_column_name>": "<key in the json file for this column>", ....} 
+    optional: True
+  script_path:
+    type: uri_file
+    description: >-
```

Comment:
Can we provide a link to the template that the user can refer to for creating their own script?

Reply:
@arun-rajora

 I have added this:

script_path:
    type: uri_file
    description: >-
      Path to the custom preprocessor python script provided by user. If both this input and template_input` are provided,
      then, `template_input` is ignored.
      The base template that should be used to create a custom preprocessor script is provided here:
      scripts/custom_dataset_preprocessors/base_preprocessor_template.py
    optional: True

Is this what you are expecting?

Reply:
Can it be a proper URL like https://github.com/Azure/azureml-assets/tree/main/assets/aml-benchmark/scripts/custom_dataset_preprocessors/base_preprocessor_template.py ? Once the PR is merged, the URL should work. Right now, it is not clear where to look for `scripts/custom_dataset_preprocessors/base_preprocessor_template.py`.

Reply:
I have added the link as suggested. Thanks @arun-rajora .

Reply:
I don't see the link. on line 31, it is still `scripts/custom_dataset_preprocessors/base_preprocessor_template.py`.

Reply:
I have not pushed that change yet. You will be able to see it once I push tem.